### DISCUSSION TITLE
Is embeddings.json intended to sync between machines?

### DISCUSSION THREAD

**Discussion by jamezpolley** [brianpetro/obsidian-smart-connections#333] [General]
I work across many different machines (and accounts on the same machine - different logins depending on whether I'm doing work for client A or client B or not working). I rely heavily on sync to keep my notes current across all the places where I work. At the moment, that means I'm using Obsidian Sync.

Is embeddings.json intended to be synced between machines? It seems like it's not, which means it's being recreated regularly - at least once on each machine. It's not a huge concern - the cost is manageable - but it seems suboptimal.

I'm not sure if this is a bug (maybe a different sync method would sync the embeddings.json better?) or if it's a feature (maybe there's something in the embeddings that ends  up being quite specific to the environment where it was created, so it really can't be synced?)

I see that one of the features for Supporters is making notes available via the ChatGPT plugin. I'm curious about how that would work - would the data uploaded to the plugin be updated every time I set up a new machine and recreate the embeddings? Or would it end up with sixteen duplicates, one copy from each machine? Would it mean that when I set up a new machine it doesn't need to recreate the local embeddings.json because it can talk to the cloud instead?

**Comment by brianpetro** [comment]
Hey @jamezpolley 

> sixteen duplicates, one copy from each machine

That's an impressive number of workstations.

You're correct in assuming it's not currently intended to be synced using the Obsidian core sync service.

The Smart Connections ChatGPT plugin does not use embeddings like the Obsidian plugin does, so the embeddings aren't synced.

In my experience, re-embedding on each machine is the easiest and cheapest compared to setting up and manage another syncing service.

Hope that helps,
Brian 🌴

**Reply by jamezpolley** [reply]
Ah! I had wondered how the embeddings would help with the chat service, but hadn't had time to look into it. Knowing that they aren't used makes a lot more sense!

If I was using the ChatGPT plugin - would the results I get from it be similar to using the Smart Connections chat inside obsidian, or are they going to be different in some way?

**Reply by jamezpolley** [reply]
This has kind of wandered off into a  "What is the current state of the chatgpt plugin" question which may not be appropriate for this forum, I just don't know of a better place to ask it.
